OsmAnd/src/net/osmand/plus/views/layers/MapTileLayer.java

```diff
@@ -39,27 +43,38 @@
 
 	public static final int OVERZOOM_IN = 2;
 	private static final int ADDITIONAL_TILE_CACHE = 3;
+	private static final int SHOWN_ZOOM_MAX = 20;
+	private static final int SHOWN_ZOOM_MIN = 1;
 
 	protected final boolean mainMap;
 	protected ITileSource map = null;
 	protected MapTileAdapter mapTileAdapter = null;
 
-	protected Paint paintBitmap;
+	protected Paint paintBitmap = null;
 	protected RectF bitmapToDraw = new RectF();
 	protected Rect bitmapToZoom = new Rect();
 
 	protected OsmandMapTileView view;
 	protected ResourceManager resourceManager;
 	protected OsmandSettings settings;
+	private boolean isSetProvider = false;
 	private boolean visible = true;
 	private boolean useSampling;
 	private StateChangedListener<Float> parameterListener;
 
+	private int getLayerId() {
```

Comment:
Better to rename to getLayerIndex and move to OsmandMapTileView

Reply:
like that idea as well instead of strange Min value

Reply:
Moved.